app/src/main/java/com/github/sdp/mediato/utility/adapters/CollectionListAdapter.java

```diff
@@ -10,30 +10,28 @@
 import androidx.annotation.Nullable;
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
-import com.github.sdp.mediato.ui.MyProfileFragment.OnAddMediaButtonClickListener;
 import com.github.sdp.mediato.R;
-import com.github.sdp.mediato.model.Review;
 import com.github.sdp.mediato.model.media.Collection;
-import com.github.sdp.mediato.utility.SampleReviews;
+import com.github.sdp.mediato.ui.MyProfileFragment.OnAddMediaButtonClickListener;
+import com.github.sdp.mediato.ui.MyProfileFragment.OnDeleteCollectionButtonClickListener;
 import java.util.List;
 
 /**
  * This adapter displays a list of collections of Media.
  */
 public class CollectionListAdapter extends RecyclerView.Adapter<CollectionListAdapter.ViewHolder> {
 
-  private Context context;
-
-  // Can be used to execute some code in the profilePage, such as updating the database
-  private OnAddMediaButtonClickListener onAddMediaButtonClickListener;
-
-  private List<Collection> collections;
+  private final Context context;
+  private final OnAddMediaButtonClickListener onAddMediaButtonClickListener;
+  private final OnDeleteCollectionButtonClickListener onDeleteCollectionButtonClickListener;

```

Comment:
Is their a reason why [`Supplier`](https://docs.oracle.com/javase/8/docs/api/java/util/function/Supplier.html), [`Function`](https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html) or [`Consumer`](https://docs.oracle.com/javase/8/docs/api/java/util/function/Consumer.html) are not used instead of these custom classes ?